src/nextjournal/clerk/parser.cljc

```diff
@@ -471,16 +470,18 @@
                                     :settings (merge-settings next-block-settings (parse-local-block-settings form))
                                     :text nstring
                                     :form (add-loc opts loc form)
-                                    :loc loc}]
-                    (when (ns? form)
+                                    :loc loc}
+                        ns? (ns? form)]
+                    (when ns?
                       (eval form))
                     (cond-> (-> state
                                 (assoc :add-comment-on-line? true)
                                 (update :nodes rest)
                                 (assoc :block-settings next-block-settings)
                                 (update :blocks conj (add-block-id code-block)))
                       (not (contains? state :ns))
-                      (assoc :ns *ns*)))
+                      (assoc :ns *ns*)
+                      ns? (update-in [:md-context :opts] merge (:nextjournal.clerk/markdown (meta *ns*)))))
```

Comment:
Think we want to also support setting this in a `nextjournal.clerk.parser/settings-marker?` which is a top-level map with one of the `nextjournal.clerk.parser/block-settings` keys set.